.github/workflows/auto-bicep.yml

```diff
@@ -0,0 +1,81 @@
+name: Auto Build Bicep
+
+on:
+  push:
+    branches-ignore:
+      - master
+    paths:
+      - '**.bicep'
+      - '!**azuredeploy.json'
+
+jobs:
+  main:
+    name: Auto Build Bicep
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 0 # avoid shallow clone so nbgv can do its work.
+
+      - name: Find sample folder
+        run: |
+          git status
+          # Get path of last file checked in
+          pwd
+          cd $GITHUB_WORKSPACE
+          pwd
+          LAST_PATH=$( git log --pretty="" --name-only -n 100 origin/master... -- | fgrep -v ".github" | head -1 )
+          echo Last file modified: $LAST_PATH
+
+          # Look for the main template file in this file's path or parents
+          SAMPLEFOLDER_PATH=$( dirname "$LAST_PATH" )
+          echo Last folder modified: $SAMPLEFOLDER_PATH
+
+          TESTFOLDER_PATH=$SAMPLEFOLDER_PATH
+          FOUNDFOLDER_PATH=
+          while [ "$TESTFOLDER_PATH" != "." ]
+          do
+            echo "Looking for main template in $TESTFOLDER_PATH"
+            MAINBICEP_PATH=$TESTFOLDER_PATH/main.bicep
+            AZDEPLOYJSON_PATH=$TESTFOLDER_PATH/azuredeploy.json
+            if [ -f "$MAINBICEP_PATH" ] || [ -f "$AZDEPLOYJSON_PATH" ]; then
+              FOUNDFOLDER_PATH=$TESTFOLDER_PATH
+              echo Found main template in $FOUNDFOLDER_PATH
+              break
+            fi
+
+            TESTFOLDER_PATH=$( dirname "$TESTFOLDER_PATH" )
+          done
+
+          if [ ! $FOUNDFOLDER_PATH ]; then
+              echo "Could not find main.bicep or azdeploy.json in folder or parents of $SAMPLEFOLDER_PATH" 1>&2
+              exit 1
+          fi
+
+          echo "SAMPLEFOLDER_PATH=$FOUNDFOLDER_PATH" >> $GITHUB_ENV
+          echo "MAINBICEP_PATH=$MAINBICEP_PATH" >> $GITHUB_ENV
+          echo "AZDEPLOYJSON_PATH=$AZDEPLOYJSON_PATH" >> $GITHUB_ENV
+
+      - name: Build main.bicep -> azuredeploy.json
+        uses: Azure/cli@v1
+        with:
+          # Specify the script here
+          inlineScript: |
+            if [ -f $MAINBICEP_PATH ]; then          
+              echo Running: az bicep build --file $MAINBICEP_PATH --outfile $AZDEPLOYJSON_PATH
+              az bicep build --file $MAINBICEP_PATH --outfile $AZDEPLOYJSON_PATH
```

Comment:
we probably need to make sure we have the latest version - the other pipelines use that so if we don't here the bicep build in the main pipeline will not match this one

Reply:
i believe that i am getting latest version... but I am going to add something to echo the versions so they are in the output.